examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -133,12 +137,17 @@ function vertical_tendency!(Yₜ, Y, cache, t)
     @. face_uₕ = Geometry.project(Geometry.Covariant12Axis(), face_u)
     @. face_uᵥ = Geometry.project(Geometry.Covariant3Axis(), face_u)
     @. Yₜ.c.ρ = -vdivf2c(Ic2f(Y.c.ρ) * face_u)
+    ᶜJ = Fields.local_geometry_field(axes(Y.c.ρ)).J
     for n in 1:5 # TODO: update RecursiveApply/Operators to eliminate this loop
         ρq_n = Y.c.ρq.:($n)
         ρqₜ_n = Yₜ.c.ρq.:($n)
         @. q_n = ρq_n / Y.c.ρ
         @. ρqₜ_n = -vdivf2c(Ic2f(ρq_n) * face_uₕ)
         if isnothing(fct_op)
+            @. ρqₜ_n -= vdivf2c(ᶠwinterp(ᶜJ, Y.c.ρ) * face_uᵥ * Ic2f(q_n))
+        elseif fct_op == upwind1
+            @. ρqₜ_n -= vdivf2c(Ic2f(Y.c.ρ) * upwind1(face_uᵥ, q_n))
```

Comment:
Is the intent to add weighted interpolation for `fct` and `zalesak` modes in a separate PR ?

Reply:
Yes; this seems reasonable.